File: modules/range-slider/ios/RangeSliderModule.swift
import ExpoModulesCore

public class RangeSliderModule: Module {
  public func definition() -> ModuleDefinition {
    Name("RangeSlider")

    // Defines event names that the module can send to JavaScript
    Events("onValueChange", "onSlidingComplete")

    // Native view component
    View(RangeSliderView.self) {
      // Props
      Prop("minValue") { (view: RangeSliderView, value: Float) in
        view.setMinValue(value)
      }
      
      Prop("maxValue") { (view: RangeSliderView, value: Float) in
        view.setMaxValue(value)
      }
      
      Prop("lowerValue") { (view: RangeSliderView, value: Float) in
        view.setLowerValue(value)
      }
      
      Prop("upperValue") { (view: RangeSliderView, value: Float) in
        view.setUpperValue(value)
      }
      
      Prop("accentColor") { (view: RangeSliderView, color: UIColor) in
        view.setAccentColor(color)
      }

      // Events
      Events("onValueChange", "onSlidingComplete")
    }
  }
}

File: modules/range-slider/android/src/main/java/expo/modules/rangeslider/RangeSliderModule.kt
package expo.modules.rangeslider

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class RangeSliderModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("RangeSlider")

    Events("onValueChange", "onSlidingComplete")

    View(RangeSliderView::class) {
      Prop("minValue") { view: RangeSliderView, value: Float ->
        view.setMin(value)
      }

      Prop("maxValue") { view: RangeSliderView, value: Float ->
        view.setMax(value)
      }

      Prop("lowerValue") { view: RangeSliderView, value: Float ->
        view.setLower(value)
      }

      Prop("upperValue") { view: RangeSliderView, value: Float ->
        view.setUpper(value)
      }

      Prop("accentColor") { view: RangeSliderView, color: Int ->
        view.setAccent(color)
      }

      Events("onValueChange", "onSlidingComplete")
    }
  }
}
